Python/ast.c

```diff
@@ -2281,6 +2281,11 @@ ast_for_print_stmt(struct compiling *c, const node *n)
     int i, j, values_count, start = 1;
 
     REQ(n, print_stmt);
+    if (Py_Py3kWarningFlag && TYPE(CHILD(n, 1)) != LPAR) {
+        if (!ast_warn(c, n,
+            "Missing parentheses in call to 'print' is not supported in 3.x"))
```

Comment:
Here's an interesting question: what should the format of 2->3 warning messages look like? From a quick glance at https://github.com/softdevteam/cpython/blob/2443fa01c96df7253fb14cb9695a2f0eb780cc8d/Lib/test/test_grammar.py, they seem to be a bit all over the shop, so we might want to define our own precedent and stick to it.

For example, some possibilities are:

* `print is a function, not a statement, in 3.x`
* `print is a function` (a bit like the `classic int division` warning)
* `print's argument(s) need to be surrounded by brackets'

I don't have a firm opinion about this: I *think* it's probably a good idea to say "3.x" in the message somewhere. Perhaps we should even use that as a warning prefix? So maybe all our warnings would be `3.x warning: print must be called as a function, not a statement.` And then we have the interesting case that *sometimes* (as in this example) we can also propose a fix. So maybe `3.x warning: print must be called as a function, not a statement. You can fix this now by surrounding print's arguments with brackets.`

Reply:
Sounds good, am just curious if we have to edit the other messages that were already there? or we can just choose to come up with our own convention for the warnings we are adding.

Reply:
I believe this PR is waiting on the answer to this query?

Reply:
Didn't you and @ltratt agree in the sync meeting the other day that we wouldn't change the existing error messages?

Reply:
> Didn't you and @ltratt agree in the sync meeting the other day that we wouldn't change the existing error messages?

Correct, we don't want to change the existing messages.